src/openrct2/actions/StaffHireNewAction.hpp

```diff
@@ -0,0 +1,345 @@
+
+        if (_staffType >= STAFF_TYPE_COUNT)
+        {
+            // Invalid staff type.
+            return MakeResult(GA_ERROR::INVALID_PARAMETERS, STR_INVALID_SELECTION_OF_OBJECTS);
+        }
+
+        if (gSpriteListCount[SPRITE_LIST_NULL] < 400)
+        {
+            return MakeResult(GA_ERROR::NO_FREE_ELEMENTS, STR_TOO_MANY_PEOPLE_IN_GAME);
+        }
+
+        if (_staffType == STAFF_TYPE_ENTERTAINER)
+        {
+            if (_entertainerType >= ENTERTAINER_COSTUME_COUNT)
+            {
+                // Invalid entertainer costume
+                return MakeResult(GA_ERROR::INVALID_PARAMETERS, STR_INVALID_SELECTION_OF_OBJECTS);
+            }
+
+            uint32_t availableCostumes = staff_get_available_entertainer_costumes();
+            if (!(availableCostumes & (1 << _entertainerType)))
+            {
+                // Entertainer costume unavailable
+                return MakeResult(GA_ERROR::INVALID_PARAMETERS, STR_INVALID_SELECTION_OF_OBJECTS);
+            }
+        }
+
+        // Look for a free slot in the staff modes.
+        int32_t staffIndex;
+        for (staffIndex = 0; staffIndex < STAFF_MAX_COUNT; ++staffIndex)
+        {
+            if (!(gStaffModes[staffIndex] & 1))
+                break;
+        }
+
+        if (staffIndex == STAFF_MAX_COUNT)
+        {
+            // Too many staff members exist already.
+            return MakeResult(GA_ERROR::NO_FREE_ELEMENTS, STR_TOO_MANY_STAFF_IN_GAME);
+        }
+
+        Peep* newPeep = &(create_sprite(GetFlags())->peep);
+        if (newPeep == nullptr)
+        {
+            // Too many staff members exist already.
+            return MakeResult(GA_ERROR::NO_FREE_ELEMENTS, STR_TOO_MANY_PEOPLE_IN_GAME);
+        }
+
+        if (execute == false)
+        {
+            // In query we just want to see if we can obtain a sprite slot.
+            sprite_remove((rct_sprite*)newPeep);
+        }
+        else
+        {
+            move_sprite_to_list((rct_sprite*)newPeep, SPRITE_LIST_PEEP * 2);
+
+            newPeep->sprite_identifier = 1;
+            newPeep->window_invalidate_flags = 0;
+            newPeep->action = PEEP_ACTION_NONE_2;
+            newPeep->special_sprite = 0;
+            newPeep->action_sprite_image_offset = 0;
+            newPeep->no_action_frame_num = 0;
+            newPeep->action_sprite_type = PEEP_ACTION_SPRITE_TYPE_NONE;
+            newPeep->path_check_optimisation = 0;
+            newPeep->type = PEEP_TYPE_STAFF;
+            newPeep->outside_of_park = 0;
+            newPeep->peep_flags = 0;
+            newPeep->paid_to_enter = 0;
+            newPeep->paid_on_rides = 0;
+            newPeep->paid_on_food = 0;
+            newPeep->paid_on_souvenirs = 0;
+
+            if (_staffType == STAFF_TYPE_HANDYMAN)
+                newPeep->staff_orders = STAFF_ORDERS_SWEEPING | STAFF_ORDERS_WATER_FLOWERS | STAFF_ORDERS_EMPTY_BINS;
+            else if (_staffType == STAFF_TYPE_MECHANIC)
```

Comment:
I think staff orders should be a param. That way people who like mowing lawn can have that.